src/main/java/io/vlingo/xoom/actors/plugin/mailbox/concurrentqueue/ConcurrentQueueMailbox.java

```diff
@@ -85,9 +72,17 @@ public boolean isSuspendedFor(String name) {
         .find(name).isEmpty();
   }
 
+  private boolean isSuspendedExceptFor(final Message override) {
+    return isSuspended() && suspendedDeliveryOverrides.get().matchesTop(override.protocol());
+  }
+
   @Override
   public Message receive() {
-    return queue.poll();
+    if (!isSuspended()) {
+      return queue.poll();
+    } else {
+      return suspendedDeliveryQueue.get().poll();
+    }
```

Comment:
I changed `receive()` to also look at the suspended delivery queue. This is a public method so I'm not 100% sure I should've done it...